testing/e2e/clients/aks.go

```diff
@@ -38,16 +40,29 @@ type aks struct {
 	options                             map[string]struct{}
 }
 
+type ServicePrincipalOptions struct {
+	ApplicationObjectID          string
+	ApplicationClientID          string
+	ServicePrincipalObjectID     string
+	ServicePrincipalCredPassword string
+}
+
+type McOptFields struct {
+	ClusterName             string
+	Ctx                     context.Context
```

Comment:
we shouldn't store ctx in a struct. Let's move it to the function definition level.

> // Programs that use Contexts should follow these rules to keep interfaces
// consistent across packages and enable static analysis tools to check context
// propagation:
//
// Do not store Contexts inside a struct type; instead, pass a Context
// explicitly to each function that needs it. The Context should be the first
// parameter, typically named ctx:
//
//	func DoSomething(ctx context.Context, arg Arg) error {
//		// ... use ctx ...
//	}

Reply:
agreed i just removed it we don't need it in any McOpts anymore, we can reintroduce it if needed